crates/refunder/src/refund_service.rs

```diff
@@ -98,63 +93,55 @@ impl RefundService {
     async fn identify_uids_refunding_status_via_web3_calls(
         &self,
         refundable_order_uids: Vec<EthOrderPlacement>,
-    ) -> Result<Vec<OrderUid>> {
+    ) -> Vec<(OrderUid, Vec<CoWSwapEthFlowAddress>)> {
         let mut batch = Web3CallBatch::new(self.web3.transport().clone());
         let futures = refundable_order_uids
             .iter()
             .map(|eth_order_placement| {
                 let order_hash: [u8; 32] = eth_order_placement.uid.0[0..32]
                     .try_into()
                     .expect("order_uid slice with incorrect length");
-                let order = self
-                    .ethflow_contract
-                    .orders(ethcontract::tokens::Bytes(order_hash))
-                    .batch_call(&mut batch);
+
+                let contract_calls = self
+                    .ethflow_contracts
+                    .iter()
+                    .map(|contract| {
+                        contract
+                            .orders(ethcontract::tokens::Bytes(order_hash))
+                            .batch_call(&mut batch)
+                    })
+                    .collect::<Vec<_>>();
+
                 async move {
-                    let order_owner = match order.await {
-                        Ok(order) => Some(order.0),
-                        Err(err) => {
-                            tracing::error!(
-                                "Error while getting the currentonchain status of orderhash {:?}, \
-                                 {:?}",
-                                H256(order_hash),
-                                err
-                            );
-                            return None;
+                    // Theoretically, multiple orders with the same order_hash can exist in multiple
+                    // contracts at the same time. That's why we need to return a list of contracts
+                    // from which the order should be refunded.
+                    let mut ethflow_contracts = vec![];
+                    for (index, call) in contract_calls.into_iter().enumerate() {
+                        match call.await {
+                            Ok((owner, _)) => {
+                                if owner != INVALIDATED_OWNER && owner != NO_OWNER {
+                                    // order has a valid owner, so we should refund it
+                                    ethflow_contracts.push(self.ethflow_contracts[index].address());
+                                }
+                            }
+                            Err(err) => {
+                                tracing::error!(
+                                    "Error while getting the currentonchain status of orderhash \
+                                     {:?}, {:?}",
+                                    H256(order_hash),
+                                    err
+                                );
```

Comment:
nit: Let's apply the usual pattern for logs while we change the code anyway.
```suggestion
                                tracing::error!(
                                    uid =? H256(order_hash),
                                    ?err,
                                    "Error while getting the current onchain status ot the order"
                                );
```